float32 target_center_x
float32 target_center_y
